### ISSUE TITLE
SRU's convergence speed is slow down in the second half of training compare to lstm

### ISSUE THREAD

**Issue by johnnykthink** [taolei87/sru#25]
Hi, I use the SRU in tensorflow for a seq2seq model.

I compare the loss with SRU and LSTM (hyper param, network structure is same), after 10 epoch, the SRU's convergence speed is slow down.
And at the 50 epoch, SRU's loss is 0.05, and LSTM's loss is only 0.01. But the validate loss is very close.
My code is here: https://github.com/johnnykthink/SRU-Tensorflow/blob/master/sru.py

So, how can I fix it? Thanks.

The training loss graph, data pair num: 300k. (blue is SRU, gray is LSTM)
![image](https://user-images.githubusercontent.com/2732807/31314039-3f1d26b4-abbb-11e7-8cf9-c894193115dd.png)

The validate loss graph, data pair num: 30k
![image](https://user-images.githubusercontent.com/2732807/31314069-150c02fe-abbc-11e7-9896-7cd814d91f76.png)

**Comment by taoleicn** [comment]
hi @johnnykthink ,

the difference on training loss shouldn't be considered as an issue, because
 - convergence on the training set depends on various factors, e.g. number of parameters, dropouts etc.
 - compared to SRU, each LSTM layer has more parameters and involves more non-linear transformation; hence given the same hyper param setting, LSTM is more likely to overfit the training data.

If you would like SRU to fit training set as LSTM does in your case, you could reduce the regularization for SRU or increase the number of parameters / layers.
